### PR TITLE
[IMP] hr_holidays: convert dashboard and widget to OWL 

### PR DESCRIPTION
**Pull request by kbapt**
Convert the `leave_stats` widget and the leaves dashboard to OWL.

TaskID: 2675380

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_holidays/static/src/dashboard/time_off_card.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            </t>
+        </span>
+    </div>
+
+    <t t-name="hr_holidays.TimeOffCardMobile" owl="1">
+        <t t-set="duration" t-value="props.requires_allocation?props.data['virtual_remaining_leaves']:props.data['virtual_leaves_taken']" />
+
+        <span class="pull-right o_timeoff_card_mobile">
+            <t t-if="props.requires_allocation" name="duration_type">
+                <strong t-esc="duration" class="o_timeoff_green"/> / <span t-esc="props.data['max_leaves']"/> <t t-if="props.data['request_unit'] == 'hour'">Hours</t><t t-else="">Days</t> <span class="o_timeoff_green">Available</span>
```
Comment (leonardopavanrocha) [inline addons/hr_holidays/static/src/dashboard/time_off_card.xml:47]:
```suggestion
                <strong t-esc="duration" class="o_timeoff_green"/> / <span t-esc="props.data.max_leaves"/> <t t-if="props.data.request_unit === 'hour'">Hours</t><t t-else="">Days</t> <span class="o_timeoff_green">Available</span>
```

File: addons/hr_holidays/static/src/dashboard/time_off_card.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            </t>
+            <t t-else="">
+                <strong t-esc="duration"/> <t t-if="props.data['request_unit'] == 'hour'">Hours</t><t t-else="">Days</t> <span class="o_timeoff_purple">Taken</span>
```
Comment (leonardopavanrocha) [inline addons/hr_holidays/static/src/dashboard/time_off_card.xml:50]:
```suggestion
                <strong t-esc="duration"/> <t t-if="props.data.request_unit === 'hour'">Hours</t><t t-else="">Days</t> <span class="o_timeoff_purple">Taken</span>
```

File: addons/hr_holidays/static/src/dashboard/time_off_card.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            </t>
+        </span>
+    </t>
+
+    <t t-name="hr_holidays.TimeOffCardPopover" owl="1">
+        <div class="o_timeoff_info">
+            <Popover position="'right'" popoverClass="'o_timeoff_popover'">
```
Comment (leonardopavanrocha) [inline addons/hr_holidays/static/src/dashboard/time_off_card.xml:57]: Do we really need to pass strings like this to OWL components? I find it ugly, but if `position="right"` doesn't work, then there's not much we can do :cry:

File: addons/hr_holidays/static/src/dashboard/time_off_card.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                <span class="fa fa-question-circle-o"/>
+                <t t-set="opened">
```
Comment (leonardopavanrocha) [inline addons/hr_holidays/static/src/dashboard/time_off_card.xml:59]: What does this `opened` variable do? I found in the codebase some usages of it, but didn't understand where it's used

File: addons/hr_holidays_attendance/static/src/xml/time_off_calendar.xml [hunk 5]
```diff
@@ -1,34 +1,30 @@
...
-        <!--inject a first condition-->
-        <xpath expr="//t[@t-if='requires_allocation']" position="before">
-            <t t-if="timeoff[1]['overtime_deductible'] == true &amp;&amp; !requires_allocation">
-                <t t-if="has_icon">
-                    <img height="30px" t-attf-src="{{timeoff[1]['icon']}}"/>
-                </t>
-                <span t-esc="timeoff[1]['virtual_remaining_leaves']" class="o_timeoff_huge o_timeoff_purple font-weight-bold align-middle"/><br/>
-                <span class="o_timeoff_purple">HOURS AVAILABLE</span>
-            </t>
+    <t t-name="hr_holidays_attendace.TimeOffCard" t-inherit="hr_holidays.TimeOffCard" t-inherit-mode="extension" owl="1">
```
Comment (leonardopavanrocha) [inline addons/hr_holidays_attendance/static/src/xml/time_off_calendar.xml:4]: Did I just find a typo :roll_eyes: ?
```suggestion
    <t t-name="hr_holidays_attendance.TimeOffCard" t-inherit="hr_holidays.TimeOffCard" t-inherit-mode="extension" owl="1">
```

**Review by leonardopavanrocha — COMMENTED** [review]
Well done :+1:  Wrote some comments/questions/suggestions but overall the PR looks good :smile:

File: addons/hr_holidays/static/src/dashboard/time_off_card.xml (hunk 3 above)
Comment (kba-odoo) [reply addons/hr_holidays/static/src/dashboard/time_off_card.xml:57]: Yep we have to do it this way :cry:

File: addons/hr_holidays/static/src/dashboard/time_off_card.xml (hunk 4 above)
Comment (kba-odoo) [reply addons/hr_holidays/static/src/dashboard/time_off_card.xml:59]: It's the slot for the body of the popover :nerd_face:  (took me a while to figure it out)
https://github.com/odoo/odoo/blob/master/addons/web/static/src/legacy/xml/base.xml#L288

**Review by xavierbol — COMMENTED** [review]
Great work ! 🥳 
I have just made two comments to avoid changing your imports when someone adds some others JS, XML or SCSS files in the module.

For the rest, LGTM. 🙂

**Comment by kba-odoo** [comment]
@robodoo r+

Thanks @leonardopavanrocha & @xavierbol for the review :kissing_heart:

**Pull request closed without merging**